src/primitives/boolean.js

```diff
@@ -0,0 +1,8 @@
+export default class BooleanState {
+  initialize(current = false) {
+    return current;
+  }
+  toggle(current = false) {
```

Comment:
I think the same applies here: transitions by their very nature, imply a current value.